src/Donations/Migrations/MoveDonationCommentToDonationMetaTable.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Give\Donations\Migrations;
+
+use Exception;
+use Give\Framework\Database\DB;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Log\Log;
+
+/**
+ * Class MoveDonationCommentToDonationMetaTable
+ *
+ * @unreleased
+ */
+class MoveDonationCommentToDonationMetaTable extends Migration
+{
+    /**
+     * @inheritdoc
+     * @throws Exception
+     */
+    public function run()
+    {
+        $commentMetaTable = DB::prefix('give_commentmeta');
+        $commentTable = DB::prefix('give_comments');
+        $donationMetaTable = DB::prefix('give_donationmeta');
+
+        $commentsCount = DB::get_var(
+            "
+            SELECT
+                COUNT(1)
+            FROM
+                $commentTable
+            WHERE
+                comment_type = 'donor_donation'
+            "
+        );
+
+        if ( ! intval($commentsCount)) {
+            return;
+        }
+
+        DB::query('START TRANSACTION');
+
+        try {
+            $insertQuery = DB::query(
+                "
+                INSERT INTO $donationMetaTable (donation_id, meta_key, meta_value)
+                SELECT
+                    comment_parent,
+                    '_give_donation_comment',
+                    comment_content
+                FROM
+                    $commentTable
+                WHERE
+                    comment_type = 'donor_donation'
+                "
+            );
+
+            if ( ! $insertQuery) {
+                throw new Exception('Failed to insert donation comment into donation meta table.');
+            }
+
+            DB::query(
+                "
+                DELETE FROM $commentTable
+                WHERE comment_type = 'donor_donation'
+                    AND comment_parent IN(
+                        SELECT
+                            donation_id FROM $donationMetaTable
+                        WHERE
+                            meta_key = '_give_donation_comment')
+                "
+            );
+
+            DB::query(
+                "
+                DELETE FROM $commentMetaTable
+                WHERE give_comment_id IN(
+                        SELECT
+                            comment_ID FROM $commentTable
+                        WHERE
+                            comment_type = 'donor_donation'
+                            AND comment_parent IN(
+                                SELECT
+                                    donation_id FROM $donationMetaTable
+                                WHERE
+                                    meta_key = '_give_donation_comment'))
+                "
+            );
+        } catch (Exception $exception) {
+            DB::query('ROLLBACK');
+
+            Log::error('Failed running migration: ' . self::title());
+
+            throw new $exception('Failed running migration: ' . self::title());
```

Comment:
It's best to throw a `DatabaseMigrationException` here. You can pass the exception caught as the third constructor argument to keep track of it.

Reply:
Resolved 1bc685a10b876c5857adb3a983089eb359facd1e